app/controllers/discover.js

```diff
@@ -364,32 +364,25 @@ exports.handleRequest = function(request, reqParams, response) {
       processData[reqParams.ajax](reqParams, render);
     } else render({ error: 'invalid call' });
   } else {
-    if (!TAB_CLASSES[reqParams.tab]) {
-      console.log('unknown tab => redirecting');
-      response.temporaryRedirect('/discover/featured'); // before: /users
-    } else if (!reqParams.loggedUser) {
-      response.temporaryRedirect('/discover/featured');
-    } else {
-      templateLoader.loadTemplate(TEMPLATE_FILE, function(template) {
-        render({
-          html: mainTemplate.renderWhydPage({
-            bodyClass: 'pgDiscover ' + TAB_CLASSES[reqParams.tab],
-            loggedUser: reqParams.loggedUser,
-            content: template.render({
-              isLogged: !!reqParams.loggedUser,
-              isAdmin: reqParams.loggedUser
-                ? request.isUserAdmin(reqParams.loggedUser)
-                : false
-            })
+    templateLoader.loadTemplate(TEMPLATE_FILE, function (template) {
+      render({
+        html: mainTemplate.renderWhydPage({
+          bodyClass: 'pgDiscover ' + TAB_CLASSES[reqParams.tab],
```

Comment:
`TAB_CLASSES[reqParams.tab]` will result to `undefined` if `reqParams.tab` is not a key of the `TAB_CLASSES` object => in order to prevents bugs, I think that we should not remove the condition from line 367.

Wouldn't removing the `else if` block from lines 370 and 371 be enough to fix this bug?